app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,75 @@
+class PasswordResetsController < ApplicationController
+  before_action :find_user, only: [:edit, :update]
+  before_action :valid_user, only: [:edit, :update]
+  before_action :check_expiration, only: [:edit, :update]
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      redirect_to_and_show_message(
+        root_url,
+        info: t("authentication.messages.reset_pwd_email_sent")
+      )
+    else
+      redirect_to_and_show_message(
+        password_resets_new_path,
+        danger: t("authentication.messages.email_not_found")
+      )
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
```

Comment:
Bạn giúp mình phân biệt empty?, blank?, present?, nil?

Reply:
empty? true nếu khôg có character 
blank? true nếu không có character hoặc full of space 
present? true nếu có ít nhất 1 ký tự khác space
nil? true nếu biến không có trong bộ nhớ

Reply:
Chỗ giải thích nil? bạn bị sai nhé. nil? chỉ trả về true nếu đó là đối tượng nil thôi nhé còn lại đều trả về false

Reply:
Chỗ này bạn đổi thành blank? vì " ".empty? => false
```suggestion
    if params[:user][:password].blank?
```